app/controllers/finance/api/payment_callbacks/stripe_callbacks_controller.rb

```diff
@@ -21,9 +21,15 @@ class MissingStripeEndpointSecret < StripeCallbackError; end
 
   # Undocumented endpoint used for update callbacks of async-authorized payment transactions (mostly due to SCA regulations)
   def create
-    service = Finance::StripePaymentIntentUpdateService.new(current_account, stripe_event)
+    service = begin
+      Finance::StripePaymentIntentUpdateService.new(current_account, stripe_event)
+    rescue ActiveRecord::RecordNotFound
+      # Returning 204 to acknowledge reception even for payments we don't care about
+      # https://issues.redhat.com/browse/THREESCALE-6851
+      nil
+    end
```

Comment:
yes, but codeclimate complains about calling return `head(:no_content)` twice

Reply:
We can approve that manually in case it gets in the way. If we agree that something is more readable or better in another way, we should do it instead of staying hostages to a tool.

Reply:
Since there's not a consensus I decided to apply this changes